apis/compute/v1alpha1/computeregiontargettcpproxy_types.go

```diff
@@ -0,0 +1,122 @@
+	// Immutable. A reference to the ComputeBackendService resource.
+	// +required
+	BackendServiceRef *refs.ComputeBackendServiceRef `json:"backendServiceRef"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Description is immutable"
+	// Immutable. An optional description of this resource.
+	Description *string `json:"description,omitempty"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Region is immutable"
+	// Immutable. The geographical location of the ComputeTargetTCPProxy.
+	// Reference: GCP definition of regions/zones (https://cloud.google.com/compute/docs/regions-zones/)
+	Region *string `json:"region"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ProxyBind is immutable"
+	// Immutable. This field only applies when the forwarding rule that references
+	// this target proxy has a loadBalancingScheme set to INTERNAL_SELF_MANAGED.
+	ProxyBind *bool `json:"proxyBind,omitempty"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ProxyHeader is immutable"
+	// Immutable. Specifies the type of proxy header to append before sending data to
+	// the backend. Default value: "NONE" Possible values: ["NONE", "PROXY_V1"].
+	ProxyHeader *string `json:"proxyHeader,omitempty"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
+	// Immutable.
+	// The ComputeRegionTargetTCPProxy name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+}
+
+// ComputeRegionTargetTCPProxyStatus defines the config connector machine state of ComputeRegionTargetTCPProxy
+type ComputeRegionTargetTCPProxyStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the ComputeRegionTargetTCPProxy resource in GCP.
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *ComputeRegionTargetTCPProxyObservedState `json:"observedState,omitempty"`
+
+	// Creation timestamp in RFC3339 text format.
+	CreationTimestamp *string `json:"creationTimestamp,omitempty"`
+
+	// Type of the resource. Always compute#targetTcpProxy for target TCP proxies.
+	Kind *string `json:"kind,omitempty"`
+
+	// The unique identifier for the resource.
+	ProxyID *int64 `json:"proxyID,omitempty"`
+
+	// The SelfLink for the resource.
+	SelfLink *string `json:"selfLink,omitempty"`
```

Comment:
since this is an Alpha resource, I assume we don't want (at least not need) these fields now?

Reply:
They are all output only fields, I'll move them to status.observedState.